modules/transaction-payment/src/lib.rs

```diff
@@ -944,6 +989,22 @@ where
 	) -> Result<Balance, DispatchError> {
 		if let Some(amount) = Self::check_native_is_not_enough(who, fee, reason) {
 			// native asset is not enough
+
+			// if override charge fee method, charge fee by the config firstly.
+			match OverrideChargeFeeMethod::<T>::get() {
+				Some(ChargeFeeMethod::FeeCurrency(fee_currency_id)) => {
+					if let Ok((_, surplus)) = Self::charge_fee_currency(who, amount, fee_currency_id) {
+						return Ok(surplus);
+					}
+				}
+				Some(ChargeFeeMethod::FeeAggregatedPath(fee_aggregated_path)) => {
+					if let Ok((_, surplus)) = Self::charge_fee_aggregated_path(who, amount, &fee_aggregated_path) {
+						return Ok(surplus);
+					}
+				}
```

Comment:
here should return an error instead of trying other tokens.
@xlc what do you think?

Reply:
yeah I agree